pallets/parachain-system/src/lib.rs

```diff
@@ -422,6 +426,8 @@ pub mod pallet {
 		/// Downward messages were processed using the given weight.
 		/// \[ weight_used, result_mqc_head \]
 		DownwardMessagesProcessed(Weight, relay_chain::Hash),
+		/// An Upward message was sent to the parent.
+		UpwardMessageSent(Option<MessageId>),
```

Comment:
Why don't you use `T::Hash` here as well?

Reply:
I just tried to mimic as much as possible what ump shows so that both ids match. But I am happy to change it if I found no issues doing so

Reply:
This really should be done as part of XCMv3, since there is a concept of XCM hashes there, and we can use it here as well.

Reply:
I can PR the xvm V3 branch if prefered

Reply:
(Keith pointed me out this SE answer when I was sniffing around for something similar: https://substrate.stackexchange.com/questions/1893/is-there-an-id-for-xcm-message/1906#1906 )

Reply:
Yes I agree with what Gav said. And although the message hash being different is something each chain can work on indepentently, I think to help explorers identify these hashes the ump queue should emit an event with this hash. I will re-work this PR to go into xcm V3, as it seems we already have the notion of "hash of the message" there